atc/atccmd/command.go

```diff
@@ -1518,14 +1577,114 @@ func (cmd *RunCommand) constructHTTPHandler(
 			// proxy Authorization header to/from auth cookie,
 			// to support auth from JS (EventSource) and custom JWT auth
 			Handler: auth.WebAuthHandler{
-				Handler: webMux,
+				Handler:    webMux,
+				Middleware: middleware,
 			},
 		},
 	}
 
 	return httpHandler
 }
 
+func (cmd *RunCommand) constructLegacyHandler(
+	logger lager.Logger,
+) (http.Handler, error) {
+	return legacyserver.NewLegacyServer(&legacyserver.LegacyConfig{
+		Logger: logger.Session("legacy"),
+	})
+}
+
+func (cmd *RunCommand) constructAuthHandler(
+	logger lager.Logger,
+	storage storage.Storage,
+) (http.Handler, error) {
+
+	issuerPath, _ := url.Parse("/sky/issuer")
+	redirectPath, _ := url.Parse("/sky/callback")
+
+	issuerURL := cmd.ExternalURL.URL.ResolveReference(issuerPath)
+	redirectURL := cmd.ExternalURL.URL.ResolveReference(redirectPath)
+
+	signingKey, err := loadOrGenerateSigningKey(cmd.Auth.AuthFlags.SigningKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add public fly client
+	cmd.Auth.AuthFlags.Clients["fly"] = "Zmx5"
```

Comment:
Can we pull this out into constants ? 
There is another ref to the client id below `validClients := []string{"fly"}`

Reply:
done